src/custom/modules/Documents/DocumentsBulkImport.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+// Shad Mickelberry
+// custom/modules/Documents/DocumentsBulkImport.php
+// 2018-11-22
+
+use Sugarcrm\Sugarcrm\Util\Uuid;
+require_once('custom/include/bulkimport/BulkImport.php');
+
+/**
+ * Class DocumentsBulkImport
+ */
+class DocumentsBulkImport extends BulkImport
+{
+
+    public function callCustomBeforeSave($b, $data, $args)
+    {
+
+        if (!empty($data['team_list'])) {
+            $teams = explode('|', $data['team_list']);
+            if (!empty($teams) && is_array($teams)) {
+                $b->team_id = $teams['0'];
+            }
+        }
+
+        // If creating a new record we need to generate an ID for the revision
+        if (!$b->id) {
+            $b->new_with_id = true;
+            $b->id = Uuid::uuid1();
+        }
+
+        // Allow for updating date modified and date entered
+        if (isset($data['date_modified'])) {
+            $b->update_date_modified = false;
+            $b->setModifiedDate($data['date_modified']);
+        }
+
+        if (isset($data['date_entered'])) {
+            $b->update_date_entered = true;
+            $b->date_entered = $data['date_entered'];
+        }
+
+        // Assign using external ID. To work the config settings for Users
+        // must also be set. Default to user->id = 1
+        if(!empty($data['external_assigned_user_key'])) {
+
+            $sugar_id = $this->getSugarRecordId(BeanFactory::getBean('Users'), $data['external_assigned_user_key']);
+            unset($b->external_assigned_user_key);
+
+            if(!empty($sugar_id)) {
+                $b->assigned_user_id = $sugar_id;
+            } else {
+                $b->assigned_user_id = 1;
+            }
+
+        } else {
+
+            $b->assigned_user_id = 1;
+
+        }
+
+        // Set created by user using external ID. To work the config settings for Users
+        // must also be set. Default to user->id = 1
+        if(!empty($data['external_created_user_key'])) {
+
+            $sugar_id = $this->getSugarRecordId(BeanFactory::getBean('Users'), $data['external_created_user_key']);
+            unset($b->external_created_user_key);
+
+            if(!empty($sugar_id)) {
+                $b->created_by = $sugar_id;
+            } else {
+                $b->created_by = 1;
+            }
+        } else {
+            $b->created_by = 1;
+        }
+
+        // Set modified by user using external ID. To work the config settings for Users
+        // must also be set. Default to user->id = 1
+        if(!empty($data['external_modified_user_key'])) {
```

Comment:
This may be the same as above. Where setting the created by and modified by users should not be in an individual hook but perhaps the core API if there it is warranted. I'll pull out for now. There isn't any greater need in Documents than other modules.

Reply:
For now I'll just leave as an example and add the flags. Please let me know your thoughts on adding to the core. I